File: 169.Majority_Element/main.py
class Solution:
    def majorityElement(self, nums):
        candidate, count = 0, 0
        for num in nums:
            if count == 0:
                candidate = num
            count += 1 if num == candidate else -1
        return candidate

# Example usage
solution = Solution()
nums1 = [3, 2, 3]
nums2 = [2, 2, 1, 1, 1, 2, 2]

print("Majority element in nums1:", solution.majorityElement(nums1)) # Output: 3
print("Majority element in nums2:", solution.majorityElement(nums2)) # Output: 2

File: 169.Majority_Element/main.ts
function majorityElement(nums: number[]): number {
     let candidate = 0, count = 0;
     for (const num of nums) {
         if (count === 0) {
             candidate = num;
         }
         count += (num === candidate) ? 1 : -1;
     }
     return candidate;
 }
 
 // Example usage
 const nums1 = [3, 2, 3];
 const nums2 = [2, 2, 1, 1, 1, 2, 2];
 
 console.log("Majority element in nums1:", majorityElement(nums1)); // Output: 3
 console.log("Majority element in nums2:", majorityElement(nums2)); // Output: 2
 